src/main/kotlin/ConsoleUIPlayer.kt

```diff
@@ -0,0 +1,12 @@
+class ConsoleUIPlayer(private val name : String, col : Int) : BasePlayer(col) {
+    override fun makeTurn(model : BaseModel): BaseTurn? {
+        println("Player $name - it's your turn.")
+        val input = readLine()
+        if (input == null)
+            return null
```

Comment:
Don't you think that "looping" should be done on the player side, and this method should return null only if the player has no possible turns?

Reply:
Fixed